crates/matrix-sdk/src/widget_api/messages/to_widget.rs

```diff
@@ -17,6 +17,18 @@ pub enum ToWidgetMessage {
     SendEvent(MessageBody<MatrixEvent, ()>),
 }
 
+impl ToWidgetMessage {
+    pub fn id(&self) -> &str {
+        match self {
+            ToWidgetMessage::SendMeCapabilities(MessageBody { header, .. })
+            | ToWidgetMessage::CapabilitiesUpdated(MessageBody { header, .. })
+            | ToWidgetMessage::OpenIdCredentials(MessageBody { header, .. })
+            | ToWidgetMessage::SendToDevice(MessageBody { header, .. })
+            | ToWidgetMessage::SendEvent(MessageBody { header, .. }) => &header.request_id,
+        }
+    }
+}
```

Comment:
This cannot be done more generically right?
There is no way to guarantee that all enum values of ToWidgetMessage contain a body with a header.
sth like: (but I think there is nothing like forcing enums to have a specific structure. I searched for this some days ago)
```
 _(MessageBody {header, ..}) => &header.request_id,
 ```
This introduces another place that needs adjusting for each added ToWidgetMessage...

Reply:
This cannot be done in a more elegant way given **the current definitions** of the `ToWidget` messages. This could be done more elegantly if we restructure `messages` a bit, namely, the common part should be extracted to the top-level definition (both messages have a header). and only the part that is different will then be part of the enum.